_release/bat/base_bat/base_bat_test.go

```diff
@@ -311,6 +327,179 @@ func TestDeleteAllInstances(t *testing.T) {
 	}
 }
 
+//Launch an instance created from an image with a customized cloud init
+//
+//Download cirros image, add it to the image service using a custom cloudinit file
+//create a workload, launch an instance of this workload and connect via SSH
+//using the custom user and password
+//
+//instance should accept connection with specified user and password and
+//reply with the user name
+func TestLaunchCustomInstance(t *testing.T) {
+	//creating a new cirros image
+	const name = "cirros-image"
+	ctx, cancelFunc := context.WithTimeout(context.Background(), standardTimeout)
+	defer cancelFunc()
+
+	// TODO:  The only options currently supported by the image service are
+	// ID and Name.  This code needs to be updated when the image service's
+	// support for meta data improves.
+	options := bat.ImageOptions{
+		Name: name,
+	}
+	//downaload cirros image
+	ImageFile, err := os.Create("cirros-0.3.4-x86_64-disk.img")
```

Comment:
Might be best to create a temporary file here, e.g.,:

f, err = ioutil.TempFile("/tmp", "ciao-random-")

We should probably delete the file as well when the test is finished.